index.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018 New Relic, Inc.  Licensed under the Apache License, version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND, either express or implied. 
+//Copyright 2018 New Relic Inc. All rights reserved.
+
+import fetch from 'node-fetch';
+import { ApiClient, RealtimeApi } from "fastly";
+
+const ACCOUNT_ID = process.env.ACCOUNT_ID
+const INSERT_KEY = process.env.INSERT_KEY
+const FASTLY_KEY = process.env.FASTLY_KEY
+const INSIGHTS_HOST = process.env.INSIGHTS_HOST || "insights-collector.newrelic.com";
+const EVENT_TYPE = process.env.EVENT_TYPE || "FastlySample";
+const POLL_INTERVAL = process.env.POLL_INTERVAL || 180000 // 3 minutes.
+
+// A limitation of Fastly is that you have to query one service at a time. I chose to create an array of my service ids, and loop through them.
+const LIST_OF_SERVICES = process.env.SERVICES.split(" ");
+
+let timestamp = 0;
+
+const fastlyApi = new ApiClient();
+fastlyApi.basePath = "https://rt.fastly.com";
+fastlyApi.authenticate(FASTLY_KEY);
+const realtimeApi = new RealtimeApi(fastlyApi);
+
+// Perform an initial poll of the services and then set an interval to poll every 3 minutes.
+pollServices();
+setInterval(pollServices, POLL_INTERVAL) // 3 minutes
+
+function pollServices() {
+  LIST_OF_SERVICES.map(async (service) => {
+    pollFromFastly(service);
+  });
+}
+
+async function pollFromFastly(service) {
+  console.log(`Polling ${service}`);
+
+  try {
+    const valuableData = await realtimeApi.getStatsLastSecond({
+      service_id: service,
+      timestamp_in_seconds: timestamp
+    });
+
+    if (valuableData.Error) {
+      console.log('error in stats', valuableData.Error)
+    } else {
+      timestamp = valuableData.Timestamp
+      valuableData.Data.map((data) => {
+        batchAndSend(data.aggregated, service)
```

Comment:
Eyeballing the data in the api docs there are a lot of keys. I didn't do a count but if there are more than 254 attributes the Event API will reject the data and drop it.

Reply:
Total attributes atm is just over 200 ⬆️